### PR TITLE
ENH: Enable filtering for ANY or NONE in ``--bids-filter-file``

### PR DESCRIPTION
**Pull request by bpinsard**
## Changes proposed in this pull request

When loading from `bids-filter-file` from json the pybids query values NONE and ANY cannot be used (cannot be serialized).

Here is a quick fix for that problem, will need to be pushed to smriprep as well, and tested.
json `null` is converted to bids.layout.Query.NONE
json string `'__any__'` is converted to bids.layout.Query.ANY (could change the naming if we prefer `'*'` but it usually means anything including empty, maybe `'+'` then ?)

## Documentation that should be reviewed

The FAQ section on how to select images when multiple one available.

### CODE REVIEW

File: fmriprep/cli/parser.py [hunk 1]
```diff
@@ -40,10 +40,19 @@ def _drop_sub(value):
         value = str(value)
         return value.lstrip('sub-')
 
+    def _filter_pybids_none_any(dct):
+        import bids
+        for k, v in dct.items():
+            if v is None:
+                dct[k] = bids.layout.Query.NONE
```
Comment (effigies) [inline fmriprep/cli/parser.py:47]: PyBIDS should already treat `None` like `Query.None`.

File: fmriprep/cli/parser.py [hunk 2]
```diff
@@ -40,10 +40,19 @@ def _drop_sub(value):
...
+            elif v == '__any__':
```
Comment (effigies) [inline fmriprep/cli/parser.py:48]: What do you think about using a wildcard?

```suggestion
            elif v == '*':
```

File: fmriprep/cli/parser.py (hunk 1 above)
Comment (bpinsard) [reply fmriprep/cli/parser.py:47]: Weird, I tried and it did not, and now it does. 
Maybe my sleep deprived brain is just making a lot of typos. :/
I will remove it.

File: fmriprep/cli/parser.py [hunk 3]
```diff
@@ -40,10 +40,17 @@ def _drop_sub(value):
         value = str(value)
         return value.lstrip('sub-')
 
+    def _filter_pybids_none_any(dct):
+        import bids
+        for k, v in dct.items():
+            if v == '__any__':
+                dct[k] = bids.layout.Query.ANY
+        return dct
```
Comment (effigies) [inline fmriprep/cli/parser.py:48]: The switch to `"*"` seems to have gotten lost in the merge.

```suggestion
        return {k: bids.layout.Query.ANY if v == "*" else v
                for k, v in dct.items()}
```

**Pull request merged**